packages/block-editor/src/components/block-caption/index.native.js

```diff
@@ -20,7 +20,7 @@ const BlockCaption = ( {
 	shouldDisplay,
 	text,
 } ) => (
-	<View style={ { flex: 1, padding: 12 } }>
+	<View style={ { flex: 1, padding: shouldDisplay ? 12 : 0 } }>
```

Comment:
I don't like that we use inline styles here, but I assume that it is like that because of the sass-transformer changes the `flex: 1` to totally different styles, right?

Reply:
>because of the sass-transformer changes the flex: 1 to totally different styles, right?

I'm not sure if it was because that or because there was only `flex` style to be applied before. 

It is transformed to 

```scss
flexGrow: 1;
flexShrink: 1;
flexBasis: 0;
```

Which seems to be expected behaviour according to [rn docs](https://reactnative.dev/docs/layout-props.html#flex)

I didn't notice any different behaviour in `BlockCaption` case after moved the `flex: 1` to stylesheet. However Luke saw the difference when working on `Buttons` and according to [this thread](https://reactnative.dev/docs/layout-props.html#flex) it seems there might be something.

Inlining `flex:1` is pattern that I have seen in codebase already so maybe it's not anti-pattern and we can go with already presented approach.

In [this comment](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1919#issuecomment-592535355) I write a note that alternative is to create the stylesheet and pull styles like below:

<details>

```diff
diff --git a/packages/block-editor/src/components/block-caption/index.native.js b/packages/block-editor/src/components/block-caption/index.native.js
index b0d00201de..9fae6e9b78 100644
--- a/packages/block-editor/src/components/block-caption/index.native.js
+++ b/packages/block-editor/src/components/block-caption/index.native.js
@@ -10,6 +10,11 @@ import { Caption, RichText } from '@wordpress/block-editor';
 import { compose } from '@wordpress/compose';
 import { withDispatch, withSelect } from '@wordpress/data';
 
+/**
+ * Internal dependencies
+ */
+import styles from './styles.scss';
+
 const BlockCaption = ( {
        accessible,
        accessibilityLabelCreator,
@@ -20,7 +25,7 @@ const BlockCaption = ( {
        shouldDisplay,
        text,
 } ) => (
-       <View style={ { flex: 1, padding: shouldDisplay ? 12 : 0 } }>
+       <View style={ [ styles.container, shouldDisplay && styles.padding ] }>
                <Caption
                        accessibilityLabelCreator={ accessibilityLabelCreator }
                        accessible={ accessible }
```

</details>

WDYT ?